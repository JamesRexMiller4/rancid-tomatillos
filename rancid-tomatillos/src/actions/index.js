export const addMovies = movies => {
  return { 
    type: 'ADD_MOVIES',
    movies
  }
}

export const addUser = id => {
  return {
    type: 'ADD_USER',
    id
  }
}

export const updateLoggedIn = isLoggedIn => {
  return {
    type: 'UPDATE_LOGGED_IN',
    isLoggedIn: !isLoggedIn
  }
}

export const removeUser = () => {
  return {
    type: 'REMOVE_USER',
    user: {}
  }
}