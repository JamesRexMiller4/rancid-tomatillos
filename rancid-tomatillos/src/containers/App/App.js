import React, { Component } from 'react';
import { addMovies } from '../../actions/index';
import { updateIsLoading } from '../../actions/index';
import './App.scss';
import Header from '../../Components/Header/Header';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import Login from '../Login/Login';
import MoviePage from '../MoviePage/MoviePage';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchMovies } from '../../apiCalls';


export class App extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount = () => {
    fetchMovies()
      .then(data => {
        this.props.addMovies(data)
        this.props.updateIsLoading(this.props.isLoading)
      })

      .catch(err => console.log(err))
  }

  render() {
    const { isLoading } = this.props
    return (
      <div className='app-container'>
        {isLoading ? <h2>Loading...</h2>
          : <>
              <Route path='/' render={({ location }) => {
                return <Header path={location.pathname}/>
                }}
              />
              <Route exact path='/' render={() => {
                return <MoviesContainer />
                }}
              />
            </>
        }
        <Route exact path='/login' render={() => {
          return <Login />
        }}
        />
        <Route exact path='/movies/:id' render={({ match }) => {
          const { id } = match.params;
          return <MoviePage id={id} />
        }}
        />
      </div>
    )
  }
};

export const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch( addMovies(movies) ),
  updateIsLoading: isLoading => dispatch( updateIsLoading(isLoading) )
});

export const mapStateToProps = state => ({
  isLoading: state.isLoading
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
