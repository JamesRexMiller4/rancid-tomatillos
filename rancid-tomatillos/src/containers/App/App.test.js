import React from 'react';
import {shallow} from 'enzyme';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { addMovies, updateIsLoading } from '../../actions';
 
describe('App', () => {
  describe('App Component', () => {
    let addMovies, updateIsLoading;
  
    beforeEach(() => {
      addMovies = jest.fn();
      updateIsLoading = jest.fn();
    });

    it('should match the snapshot with all the data passed through', () => {
      const wrapper = shallow(<App 
          addMovies={addMovies}
          updateIsLoading={updateIsLoading}
          isLoading={true}
        />)
  
  
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a different snapshot when isLoading is false', () => {
        const wrapper = shallow(<App 
          addMovies={addMovies}
          updateIsLoading={updateIsLoading}
          isLoading={false}
        />)
  
        expect(wrapper).toMatchSnapshot();
      });
  });
 
  describe('mapDispatchToProps', () => {
    it('should call dispatch with addMovies', () => {
      const movies = [{
          average_rating: 6,
          backdrop_path: "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
          id: 1,
          overview: "In Jumanji: The Next Level",
          poster_path: "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
          release_date: "2019-12-04",
          title: "Jumanji: The Next Level",
        },
        {
          average_rating: 5.666666666666667,
          backdrop_path: "https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          id: 2,
          overview: "The near future",
          poster_path: "https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          release_date: "2019-09-17",
          title: "Ad Astra",
        }];

      const mockDispatch = jest.fn();
      const actionToDispatch = addMovies(movies);
      
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addMovies(movies)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with updateIsLoading', () => {
      const mockDispatch = jest.fn();
      const mockLoading = true;
      const actionToDispatch = updateIsLoading(mockLoading);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.updateIsLoading(mockLoading);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  describe('mapStateToProps', () => {
    it('should return isLoading from the store', () => {
      const mockState = {
        isLoading: true,
        isLoggedIn: false
      };
  
      const expected = {
        isLoading: true
      };
  
      const mappedState = mapStateToProps(mockState);
  
      expect(mappedState).toEqual(expected);
    });
  });
});
