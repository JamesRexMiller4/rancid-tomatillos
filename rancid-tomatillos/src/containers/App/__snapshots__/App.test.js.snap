// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot with all the data passed through 1`] = `
<div
  className="app-container"
>
  <h2>
    Loading...
  </h2>
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
</div>
`;

exports[`App App Component should render a different snapshot when isLoading is false 1`] = `
<div
  className="app-container"
>
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
</div>
`;
